controllers/etcdcluster_controller_test.go

```diff
@@ -56,10 +63,149 @@ type StaticResponseEtcdAPI struct {
+
+		// Make our fake etcd respond
+		s.etcd = fakeEtcdForEtcdCluster(*etcdCluster)
+
+		t.Log("Given a 3-node cluster")
+		err := s.k8sClient.Create(s.ctx, etcdCluster)
+		require.NoError(t, err, "failed to create EtcdCluster resource")
+
+		clusterObjectKey := client.ObjectKey{
+			Namespace: namespace,
+			Name:      etcdCluster.Name,
+		}
+
+		t.Log("Which has all its nodes")
+		err = try.Eventually(func() error {
+			var updatedCluster etcdv1alpha1.EtcdCluster
+			err := s.k8sClient.Get(s.ctx, clusterObjectKey, &updatedCluster)
+			require.NoError(t, err, "failed to find EtcdCluster")
+			if updatedCluster.Status.Replicas != *etcdCluster.Spec.Replicas {
+				return fmt.Errorf(
+					"unexpected replicas count in cluster status. replicas: %d, status: %#v",
+					updatedCluster.Status.Replicas, updatedCluster.Status,
+				)
+			}
+			return nil
+		}, time.Second*5, time.Millisecond*500)
+		require.NoError(t, err)
+
+		t.Log("When the cluster is scaled down to 1-node")
+		patch := client.MergeFrom(etcdCluster.DeepCopy())
+		*etcdCluster.Spec.Replicas = 1
+		err = s.k8sClient.Patch(s.ctx, etcdCluster, patch)
+		require.NoError(t, err, "failed to patch EtcdCluster resource")
+
+		t.Run("StatusUpdate", func(t *testing.T) {
+			t.Log("The cluster status is updated when the cluster has been scaled down.")
+			expectedStatus := expectedStatusForCluster(*etcdCluster)
+			err = try.Eventually(func() error {
+				var updatedCluster etcdv1alpha1.EtcdCluster
+				err := s.k8sClient.Get(s.ctx, client.ObjectKey{
+					Namespace: namespace,
+					Name:      etcdCluster.Name,
+				}, &updatedCluster)
+				require.NoError(t, err, "failed to find EtcdCluster")
+
+				if diff := cmp.Diff(expectedStatus, updatedCluster.Status); diff != "" {
+					return fmt.Errorf("unexpected EtcdCluster.Status: diff --- expected, +++ actual\n%s", diff)
+				}
+				return nil
+			}, time.Second*20, time.Millisecond*500)
+			assert.NoError(t, err)
+		})
+		t.Run("EtcdMembersUpdate", func(t *testing.T) {
+			t.Log("The etcd cluster API reports the expected number of nodes")
+			membership, err := s.etcd.MembershipAPI(etcdclient.Config{})
+			require.NoError(t, err)
+			expectedMembers := expectedEtcdMembersForCluster(*etcdCluster)
+			err = try.Eventually(func() error {
+				members, err := membership.List(s.ctx)
+				require.NoError(t, err)
+				if diff := cmp.Diff(expectedMembers, members); diff != "" {
+					return fmt.Errorf("unexpected Etcd API members: diff --- expected, +++ actual\n%s", diff)
+				}
+				return nil
+			}, time.Second*5, time.Millisecond*500)
+			assert.NoError(t, err)
+		})
+		t.Run("EtcdPeersRemoved", func(t *testing.T) {
+			t.Log("The EtcdPeers are removed")
+			fieldsOfInterest := []string{
+				".metadata.name",
+				".metadata.namespace",
+			}
+			expectedPeers := expectedEtcdPeersForCluster(*etcdCluster)
+			expectedValues, err := test.ExtractStructFieldsFromSlice(fieldsOfInterest, peersToInterfaces(expectedPeers))
```

Comment:
On reflection I might have got carried away with  `ExtractStructFieldsFromSlice`.
I just want to check the .metatdata.name of the items in the  `EtcdPeerList`, ignoring all the other fields.